src/PostgREST/Auth/JwtCache.hs

```diff
@@ -25,19 +27,30 @@ import PostgREST.Error      (Error (..))
 
 import Protolude
 
-newtype JwtCacheState = JwtCacheState
-  { jwtCache :: C.Cache ByteString AuthResult
+-- | JWT Cache and IO action that triggers purging old entries from the cache
+data JwtCacheState = JwtCacheState
+  { jwtCache   :: C.Cache ByteString AuthResult
+  , purgeCache :: IO ()
   }
 
 -- | Initialize JwtCacheState
 init :: IO JwtCacheState
 init = do
   cache <- C.newCache Nothing -- no default expiration
-  return $ JwtCacheState cache
+  -- purgeExpired has O(n^2) complexity
+  -- so we wrap it in debounce to make sure it:
+  -- 1) is executed asynchronously
+  -- 2) only a single purge operation is running at a time
+  debounce <- mkDebounce defaultDebounceSettings
+    -- debounceFreq is set to default 1 second
+    { debounceAction = C.purgeExpired cache
```

Comment:
Are we sure this debounceAction is targeting the up-to-date jwt cache? Does `newCache` create an `IORef`?

Reply:
Ok it's a `TVar`([ref](https://hackage.haskell.org/package/cache-0.1.3.0/docs/src/Data.Cache.html#newCache)), so it is a mutable reference like `IORef`.